flows/urns.go

```diff
@@ -27,8 +28,78 @@ func ValidateURNScheme(fl validator.FieldLevel) bool {
 	return urns.IsValidScheme(fl.Field().String())
 }
 
+type ContactURN struct {
+	urns.URN
+	channel Channel
+}
+
+func (u *ContactURN) Channel() Channel { return u.channel }
+
+// Resolve is called when a URN is part of an excellent expression
+func (u *ContactURN) Resolve(key string) interface{} {
+	switch key {
+	case "scheme":
+		return u.URN.Scheme()
+	case "path":
+		return u.URN.Path()
+	case "display":
+		return u.URN.Display()
+	case "channel":
+		return u.Channel()
+	}
+	return fmt.Errorf("no field '%s' on URN", key)
+}
+
+func (u *ContactURN) Default() interface{} { return u }
+
+func (u *ContactURN) String() string { return string(u.URN) }
+
 // URNList is the list of a contact's URNs
-type URNList []urns.URN
+type URNList []*ContactURN
```

Comment:
Why pointer?

Reply:
Because I have an irrational fear of passing around structs that I need to get over?